### ISSUE TITLE
Access Validation Data in on_epoch_finished

### ISSUE THREAD

**Issue by BrianMiner** [dnouri/nolearn#94]
Is it possible to access the validation X and y (from eval parameter) within a function passed to on_epoch_finished? I would like to either print out or use early stopping for a metric that is not the objective loss function? Thus I was looking for predicting the y from the validation set at each epoch and calculating a this metric.

**Comment by dnouri** [comment]
You should be able to use the (undocumented) `custom_score` parameter for this.  When you instantiate the NeuralNet, pass something like:
```python
NeuralNet(
    ...
    custom_score=('myscore', myscore_func),
    ...
    )
```

You should then have a `myscore` inside every entry of `train_history`.

**Comment by BenjaminBossan** [comment]
Also, I think that
```
Xt, Xv, yt, yv = net.train_test_split(X, y, net.eval_size)
```
should result in the same train/validation split as used during training

**Comment by BrianMiner** [comment]
Thanks for the replies, I have several options it seems. I stumbled upon this which I think (hope) also worked (I am doing this within the EarlyStopping class _call_):

```nn.batch_iterator_test.y ```    <-I think this is the true y of the validation sample
```nn.predict(nn.batch_iterator_test.X)```  <- I think this is the X of the validation sample and I am predicting  with the current model

**Comment by dnouri** [comment]
Looks right!
